parachain-tracer/src/tracker.rs

```diff
@@ -497,25 +563,31 @@ impl SubxtTracker {
 		time_diff(self.current_relay_block.map(|v| v.ts), self.on_demand_order_at.map(|v| v.ts))
 	}
 
-	fn has_backed_candidate(&self) -> bool {
-		self.current_candidate.candidate.is_some() ||
+	fn has_backed_candidate(&self, core: u32) -> bool {
+		self.candidates
+			.get(&core)
+			.map_or(false, |v| v.iter().any(|candidate| candidate.candidate_hash.is_some())) ||
 			self.relay_forks
 				.iter()
 				.any(|fork| fork.backed_candidate.is_some() || fork.included_candidate.is_some())
 	}
 
+	fn is_current_candidate_backed(&self, core: u32) -> bool {
+		self.current_candidate(core).map_or(false, |v| v.is_backed())
+	}
+
 	fn is_just_backed(&self) -> bool {
 		self.last_backed_at_block_number.is_some() &&
 			self.last_backed_at_block_number == self.current_relay_block.map(|v| v.num)
 	}
 
-	fn is_slow_availability(&self) -> bool {
-		self.current_candidate.core_occupied &&
+	fn is_slow_availability(&self, core: u32) -> bool {
```

Comment:
I think it is a good time to make sure we have tests covering all of these functions and ensure they work as expected. The condition below seems brittle.

Reply:
It's covered here
https://github.com/paritytech/polkadot-introspector/blob/ae52a2c72e92d920b8d4b84d15974219e3ae8ce9/parachain-tracer/src/tracker.rs#L1298-L1319